### PR TITLE
Update faster route with list of routes

### PR DESCRIPTION
**Pull request by kmadsen**
## Description

Following up from https://github.com/mapbox/mapbox-navigation-android/pull/2939#issuecomment-625995421

@JunDai noticed that accepting faster route is not straight forward. Proposing an api change to make it so faster routes can be accepted without any List manipulation.

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [ ] I have added the appropriate milestone and project boards

## Testing

Please describe the manual tests that you ran to verify your changes

- [ ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [ ] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Checklist

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: examples/src/main/java/com/mapbox/navigation/examples/core/SimpleMapboxNavigationKt.kt [hunk 1]
```diff
@@ -247,12 +247,9 @@ class SimpleMapboxNavigationKt : AppCompatActivity(), OnMapReadyCallback,
...
             fasterRouteSelectionTimer.onFinish()
         }
-        acceptLayout.setOnClickListener {
-            fasterRoute?.let {
-                mapboxNavigation.setRoutes(mapboxNavigation.getRoutes().toMutableList().apply {
-                    removeAt(0)
-                    add(0, it)
-                })
+        acceptLayout.setOnClickListener { _ ->
+            fasterRoutes.takeIf { it.isNotEmpty() }?.let { newRoutes ->
```
Comment (kmadsen) [inline examples/src/main/java/com/mapbox/navigation/examples/core/SimpleMapboxNavigationKt.kt:251]: nit: `fasterRoutes.takeIf { it.isNotEmpty() }?.let { newRoutes ->`

better book-keeping. wonder if core can/should help with this

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/fasterroute/FasterRouteObserver.kt [hunk 2]
```diff
@@ -16,11 +16,11 @@ interface FasterRouteObserver {
...
      *
-     * @param currentRoute reference to the current route.
-     * @param alternativeRoute reference to the alternative checked.
-     * @param isAlternativeFaster true if the alternativeRoute is faster, false otherwise
+     * @param currentRoute the current route.
+     * @param alternatives the routes where alternatives[0] may be faster
+     * @param isAlternativeFaster true if the alternatives[0] is faster, false otherwise
      */
-    fun onFasterRoute(currentRoute: DirectionsRoute, alternativeRoute: DirectionsRoute, isAlternativeFaster: Boolean)
+    fun onFasterRoute(currentRoute: DirectionsRoute, alternatives: List<DirectionsRoute>, isAlternativeFaster: Boolean)
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/fasterroute/FasterRouteObserver.kt:23]: For documentation purposes noting that this breaks SemVer cc @langsmith

File: examples/src/main/java/com/mapbox/navigation/examples/core/SimpleMapboxNavigationKt.kt (hunk 1 above)
Comment (Guardiola31337) [reply examples/src/main/java/com/mapbox/navigation/examples/core/SimpleMapboxNavigationKt.kt:251]: Look at this `TODO` / question 😅 https://github.com/mapbox/mapbox-navigation-android/blob/3fd77323074d64217606f85291cd38b9f51297d5/libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt#L113-L114

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/fasterroute/FasterRouteObserver.kt [hunk 3]
```diff
@@ -16,11 +16,11 @@ interface FasterRouteObserver {
     /**
      * Invoked whenever a faster route was inspected.
```
Comment (langsmith) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/fasterroute/FasterRouteObserver.kt:17]:
```suggestion
     * Invoked whenever a faster route is available.
```

**Comment by langsmith** [comment]
Change `FasterRouteObserver`'s docs

From

```
 * Interface definition for an observer that get's notified whenever the SDK finds a faster route to the destination.
```

To

```
 * Interface definition for an observer that notifies whenever the Navigation SDK finds a faster route to the destination.
```

File: examples/src/main/java/com/mapbox/navigation/examples/core/SimpleMapboxNavigationKt.kt (hunk 1 above)
Comment (kmadsen) [reply examples/src/main/java/com/mapbox/navigation/examples/core/SimpleMapboxNavigationKt.kt:251]: yeah we rely on list order in a lot of places now. another option is to add a function that does something like this

https://github.com/mapbox/mapbox-navigation-android/blob/80d5359f2abb6d82bd5613b077bdbc3902b63143/examples/src/main/java/com/mapbox/navigation/examples/core/FasterRouteActivity.kt#L241-L244

``` kotlin
MapboxNavigation

    fun setRoutes(routes: List<DirectionsRoute>) {
        directionsSession.routes = routes
    }
    fun setRoute(route: DirectionsRoute) {
        setRoutes(mapboxNavigation.getRoutes().toMutableList().apply {
            removeAt(0)
            add(0, it)
        })
    }

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/fasterroute/FasterRouteObserver.kt (hunk 2 above)
Comment (langsmith) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/fasterroute/FasterRouteObserver.kt:23]: Thanks @Guardiola31337

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/fasterroute/FasterRouteObserver.kt (hunk 3 above)
Comment (kmadsen) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/fasterroute/FasterRouteObserver.kt:17]: the callback happens even if the result was full of slower routes, so "whenever faster route is available" doesn't feel accurate 🤔

**Pull request merged**